app/models/test_run.rb

```diff
@@ -109,6 +115,33 @@ def create_test_group_result(result)
     )
   end
 
+  def add_extra_marks(extra_marks)
+    extra_marks_total = 0
+    submission_result = self.grouping&.current_submission_used&.get_latest_result
```

Comment:
Use `self.submission`, as we do in the `update_results!` method.

Reply:
Hi @Naragod, I'm saying you can use `self.submission` instead of `self.grouping&.current_submission_used` (`self.submission` might still be `nil`, and using `&.` after that is totally fine).